hoomd/device.py

```diff
@@ -67,264 +54,320 @@ def __init__(self, communicator, notice_level, msg_file, shared_msg_file):
 
     @property
     def comm(self):
-        """
-        Get the MPI Communicator
-        """
-
+        warnings.warn("Use communicator.", DeprecationWarning)
         return self._comm
 
-    # \brief Return the name of the GPU used in GPU mode.
     @property
-    def gpu_ids(self):
-
-        n_gpu = self.cpp_exec_conf.getNumActiveGPUs()
-        return [self.cpp_exec_conf.getGPUName(i) for i in range(n_gpu)]
+    def communicator(self):
+        """comm.Communicator: The MPI Communicator [read only]."""
+        return self._comm
 
-    # \brief Return the execution mode
     @property
     def mode(self):
```

Comment:
Given that everything device will be a `GPU` or `CPU` device do we need this any more?

Reply:
No we do not. For now, `device.mode` is used heavily throughout the codebase, we can clean it up later after we get more PRs merged: #755

Reply:
nvm, will just make the change now and other PRs will need to adapt before merging.